user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -121,24 +149,50 @@ static void get_new_data(Gimbal_t *gimbal_data){
  */
 static void update_setpoints(Gimbal_t *gimbal_set){
     
-    gimbal_set->yaw_motor.pos_set += (-1) * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND) / 10;
-    gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 10;
+    if(gimbal_set->rc_update->rc.s[0] == RC_SW_MID || gimbal_set->rc_update->rc.s[0] == RC_SW_UP){
+        fp32 theta = -1 * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND)
+                * MOTOR_ECD_TO_RAD / 80.0f;
+        fp32 rotation[2] = {arm_cos_f32(theta), arm_sin_f32(theta)};
+        multiply_complex_a_by_b(gimbal_set->yaw_setpoint, rotation);
+        make_unit_length(gimbal_set->yaw_setpoint);
+
+        gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 80.0f;
+    }
     
-    int16_constrain(gimbal_set->yaw_motor.pos_set, YAW_MIN, YAW_MAX);
-    int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+    gimbal_set->pitch_motor.pos_set = int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
```

Comment:
The int16_deadzone/constrain() functions could be made into macros DEADZONE and CONSTRAIN - feels cleaner than the explicit type

Reply:
After trying to implement this with macros and reading more this seems to be a bad idea. 
1. Macro is purely a side-effect 
2. Macro must use ternary statement (no-ifs) so and this leads to a problem in the default case
3. Types are limited in C, we should not strive to create *more* ambiguity.
`#define DEADZONE(num, lower_bound, upper_bound) ({num < upper_bound && num > lower_bound ? num = 0:true})
#define CONSTRAIN(num, lower_bound, upper_bound) ({(num < lower_bound ? num = lower_bound : \
            (num > upper_bound ? num = upper_bound : <???>)})`

Reply:
Sure, macros can have side effects if misused and the int16 prefix does remind us of the type, so we could stick with it. For all intents and purposes this is style thing - the prefix doesn't happen to be my style.

2. No, they can. See pid.c